src/pins/pin.rs

```diff
@@ -236,9 +249,11 @@ where
     /// [`GPIO`]: ../gpio/struct.GPIO.html
     pub fn into_input_pin(
         self,
-        token: Token<T, init_state::Enabled>,
+        _token: Token<T, init_state::Enabled>,
     ) -> GpioPin<T, direction::Input> {
-        GpioPin::new(token, ())
+        // note that `_token` is consumed and discarded at this pint because we don't need it
+        // anymore– it has served its purpose of guaranteeing pin uniqueness
```

Comment:
Alternative suggestion:

```suggestion
        // note that `_token` is consumed and discarded at this pint because we don't need it
        // anymore– it has served its purpose
```